### PR TITLE
fix(roxvet): generted file comment checker

### PR DESCRIPTION
**Pull request by janisz**
## Description

The generated comment does not need to be on the top of the file as per doc:

> [This line must appear before the first non-comment, non-blank text in the file.](https://pkg.go.dev/cmd/go#hdr-Generate_Go_files_by_processing_source:~:text=This%20line%20must%20appear%20before%20the%20first%20non%2Dcomment%2C%20non%2Dblank%20text%20in%20the%20file.)

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

### Here I tell how I validated my change

Run it on the https://github.com/stackrox/stackrox/pull/10379 and it worked previously scanner files were reported as they have a comment before generated comment

### Reminder for reviewers

In addition to reviewing code here, reviewers **must** also review testing and request further testing in case the
performed one does not seem sufficient. As a reviewer, you must not approve the change until you understand the
performed testing and you are satisfied with it.

### CODE REVIEW

**Comment by janisz** [comment]
* **#10378** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/10378?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @janisz and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: tools/roxvet/common/file_filters.go [hunk 1]
```diff
@@ -15,14 +15,14 @@ func IsGeneratedFile(f *ast.File) bool {
...
 	}
-	firstComment := f.Comments[0].List[0]
-
-	if firstComment.Pos() > f.Package {
-		// comment is not at beginning of file, before the package keyword
-		return false
+	for _, comments := range f.Comments {
+		for _, comment := range comments.List {
+			if generatedCodeRegex.MatchString(comment.Text) {
+				return true
```
Comment (mtodor) [inline tools/roxvet/common/file_filters.go:21]: Should we still keep checking that it is before the package?
```suggestion
				return comment.Pos() < f.Package
```


Another option is to check it before `if` and return if the comment position is after `f.Package`. That would also break the loop earlier. But we can do that only if we are sure that the order of comments is correct.

**Review by rhybrillou — APPROVED** [review]
This looks good.
Ideally, we should write unit tests with some basic file payload similar to what is available in `traverse_test.go`.

File: tools/roxvet/common/file_filters.go [hunk 2]
```diff
@@ -15,14 +15,15 @@ func IsGeneratedFile(f *ast.File) bool {
...
 		return false
 	}
-	firstComment := f.Comments[0].List[0]
-
-	if firstComment.Pos() > f.Package {
-		// comment is not at beginning of file, before the package keyword
-		return false
+	for _, comments := range f.Comments {
+		for _, comment := range comments.List {
+			if generatedCodeRegex.MatchString(comment.Text) {
```
Comment (dhaus67) [inline tools/roxvet/common/file_filters.go:20]: nit: also let's make sure we only take into account comments which are _before_ `f.Package` and short-circuit here if they are not.

File: tools/roxvet/common/file_filters.go [hunk 3]
```diff
@@ -15,14 +15,15 @@ func IsGeneratedFile(f *ast.File) bool {
 	if len(f.Comments) == 0 {
```
Comment (dhaus67) [inline tools/roxvet/common/file_filters.go:15]: don't need the short-circuit here since we use `for range` below to iterate over the list of comments.

**Review by dhaus67 — COMMENTED** [review]
Let's also add some tests to this, there's `tools/roxvet/common/traverse_test.go` which can be used as inspiration.

**Pull request merged**